design-docs/operationBased2.md

```diff
@@ -0,0 +1,269 @@
+## Example 1
+
+<table>
+<tr><th>GraphQL</th><th>Kotlin</th></tr>
+<tr>
+<td><pre lang="graphql">
+query TestOperation {
+  animal {
+    species
+  }
+}
+</pre></td>
+<td><pre lang="kotlin">
+public data class Data(public val animal: Animal?)
+data class Animal(val species: String)
+</pre></td>
+</tr>
+</table>
+
+
+## Example 2
+
+<table>
+<tr><th>GraphQL</th><th>Kotlin</th></tr>
+<tr>
+<td><pre lang="graphql">
+query TestOperation {
+  animal {
+    ... LionFragment
+    ... CatFragment
+  }
+}
+fragment LionFragment on Cat {
+  species
+}
+fragment CatFragment on Cat {
+  species
+}
+</pre></td>
+<td><pre lang="kotlin">
+public data class Data(public val animal: Animal?)
+data class Animal(
+  val lionFragment: LionFragment?,
+  val catFragment: CatFragment?,
+)
+</pre></td>
+</tr>
+</table>
+
+
+## Example 3
+
+<table>
+<tr><th>GraphQL</th><th>Kotlin</th></tr>
+<tr>
+<td><pre lang="graphql">
+query TestOperation {
+  animal {
+    # Field type = Inline fragment type
+    ... on Animal {
+      species
+    }
+  }
+}
+</pre></td>
+<td><pre lang="kotlin">
+public data class Data(public val animal: Animal?)
+data class Animal(val onAnimal: OnAnimal)
+</pre></td>
+</tr>
+</table>
+
+## Example 4
+
+<table>
+<tr><th>GraphQL</th><th>Kotlin</th></tr>
+<tr>
+<td><pre lang="graphql">
+query TestOperation {
+  cat {
+    # Field type Cat is subtype of inline fragment
+    ... on Animal {
+      species
+    }
+  }
+}
+</pre></td>
+<td><pre lang="kotlin">
+public data class Data(public val cat: Cat?)
+data class Cat(val onAnimal: OnAnimal)
+</pre></td>
+</tr>
+</table>
+
+## Example 5
+
+<table>
+<tr><th>GraphQL</th><th>Kotlin</th></tr>
+<tr>
+<td><pre lang="graphql">
+query TestOperation {
+  animal {
+    ... on Cat {
+      species
+    }
+  }
+}
+</pre></td>
+<td><pre lang="kotlin">
+public data class Data(public val animal: IAnimal?)
+
+sealed interface IAnimal { val onCat: Animal.OnCat? }
+data class CatAnimal(val onCat: Animal.OnCat): IAnimal
+data class Animal(val onCat: Animal.OnCat?): IAnimal
```

Comment:
For consistency with responseBased codegen:

```suggestion
sealed interface Animal { val onCat: Animal.OnCat? }
data class CatAnimal(val onCat: Animal.OnCat): Animal
data class OtherAnimal(val onCat: Animal.OnCat?): Animal
```

Reply:
OK. 

At first, I thought that it would be more compatible with `operationBased` if I named classes like these. But I was wrong. Your suggestion is more compatible with `operationBased` codegen.